src/libraries/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -813,13 +813,43 @@ private static void WriteSetColorString(bool foreground, ConsoleColor color)
         /// <summary>Writes out the ANSI string to reset colors.</summary>
         private static void WriteResetColorString()
         {
-            // We only want to send the reset string if we're targeting a TTY device
-            if (!Console.IsOutputRedirected)
+            if (SupportsAnsiColor())
             {
                 WriteStdoutAnsiString(TerminalFormatStrings.Instance.Reset);
             }
         }
 
+        /// <summary>
+        /// Tests whether ANSI color codes should be emitted or not.
+        /// <para>
+        /// If the <c>DOTNET_CONSOLE_ANSI_COLOR</c> environment variable contains <c>true</c> (case insensitive) or <c>1</c>
+        /// then ANSI color codes are supported. If the <c>DOTNET_CONSOLE_ANSI_COLOR</c> environment variable is not defined
+        /// or contains a non truthy value, then ANSI color codes are supported if the console output is not redirected.
+        /// </para>
+        /// </summary>
+        /// <returns><c>true</c> if ANSI color escape codes must be emitted, <c>false</c> if they must not be emitted.</returns>
+        /// <remarks>This was discussed in https://github.com/dotnet/runtime/issues/33980</remarks>
+        private static bool SupportsAnsiColor()
+        {
+            string? consoleAnsiColor = Environment.GetEnvironmentVariable("DOTNET_CONSOLE_ANSI_COLOR");
+            if (consoleAnsiColor != null)
+            {
+                return IsTrueStringIgnoreCase(consoleAnsiColor) || consoleAnsiColor.Equals("1");
+            }
+
+            return !Console.IsOutputRedirected;
+        }
+
+        // Copied from System.Boolean (where it's internal)
+        private static bool IsTrueStringIgnoreCase(string value)
+        {
+            return value.Length == 4 &&
+                   (value[0] == 't' || value[0] == 'T') &&
+                   (value[1] == 'r' || value[1] == 'R') &&
+                   (value[2] == 'u' || value[2] == 'U') &&
+                   (value[3] == 'e' || value[3] == 'E');
+        }
```

Comment:
```suggestion
        private static bool SupportsAnsiColor()
        {
            string? consoleAnsiColor = Environment.GetEnvironmentVariable("DOTNET_CONSOLE_ANSI_COLOR"); 
            if (consoleAnsiColor != null)
            {
                return consoleAnsiColor == "1" || (bool.TryParse(consoleAnsiColor, out bool enabled) && enabled);
            }

            return !Console.IsOutputRedirected;
        }
```

Reply:
That was actually how I wrote it first. Then I searched in the dotnet runtime repository to see if there was an idiomatic way of checking if an environment variable is *true or 1*. The first one I found was in the [Globalization subsystem](https://github.com/dotnet/runtime/blob/v5.0.2/src/libraries/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.cs#L24).

But I'm happy to not duplicate the `IsTrueStringIgnoreCase` method once more, so I applied your suggestion.